core/policy/policy_test.go

```diff
@@ -162,31 +162,72 @@ func TestPolicyChecker(t *testing.T) {
-	err = pc.CheckPolicy("C", "readers", sProp)
-	require.Error(t, err)
-	require.Contains(t, err.Error(), "Failed evaluating policy on signed data during check policy on channel [C] with policy [readers]: [Invalid Signature]")
-
-	// Alice is a member of the local MSP, policy check must succeed
-	identityDeserializer.Msg = sProp.ProposalBytes
-	err = pc.CheckPolicyNoChannel(mgmt.Members, sProp)
-	require.NoError(t, err)
-
-	sProp, _ = protoutil.MockSignedEndorserProposalOrPanic("A", &peer.ChaincodeSpec{}, []byte("Bob"), []byte("msg2"))
-	// Bob is not a member of the local MSP, policy check must fail
-	err = pc.CheckPolicyNoChannel(mgmt.Members, sProp)
-	require.Error(t, err)
-	require.Contains(t, err.Error(), "Failed deserializing proposal creator during channelless check policy with policy [Members]: [Invalid Identity]")
+	t.Run("CheckPolicy", func(t *testing.T) {
+		// Validate Alice signatures against channel A's readers
+		sProp, _ := protoutil.MockSignedEndorserProposalOrPanic("A", &peer.ChaincodeSpec{}, []byte("Alice"), []byte("msg1"))
+		policyManagerGetter.Managers["A"].(*mocks.MockChannelPolicyManager).MockPolicy.(*mocks.MockPolicy).Deserializer.(*mocks.MockIdentityDeserializer).Msg = sProp.ProposalBytes
+		sProp.Signature = sProp.ProposalBytes
+		err := pc.CheckPolicy("A", "readers", sProp)
+		require.NoError(t, err)
+
+		// Proposal from Alice for channel A should fail against channel B, where Alice is not involved
+		err = pc.CheckPolicy("B", "readers", sProp)
+		require.Error(t, err)
+		require.Contains(t, err.Error(), "Failed evaluating policy on signed data during check policy on channel [B] with policy [readers]: [Invalid Identity]")
+
+		// Proposal from Alice for channel A should fail against channel C, where Alice is involved but signature is not valid
+		err = pc.CheckPolicy("C", "readers", sProp)
+		require.Error(t, err)
+		require.Contains(t, err.Error(), "Failed evaluating policy on signed data during check policy on channel [C] with policy [readers]: [Invalid Signature]")
+	})
+
+	t.Run("CheckPolicyNoChannel", func(t *testing.T) {
+		sProp, _ := protoutil.MockSignedEndorserProposalOrPanic("A", &peer.ChaincodeSpec{}, []byte("Alice"), []byte("msg1"))
+		sProp.Signature = sProp.ProposalBytes
+
+		// Alice is a member of the local MSP, policy check must succeed
+		identityDeserializer.Msg = sProp.ProposalBytes
+		err := pc.CheckPolicyNoChannel(mgmt.Members, sProp)
+		require.NoError(t, err)
+
+		sProp, _ = protoutil.MockSignedEndorserProposalOrPanic("A", &peer.ChaincodeSpec{}, []byte("Bob"), []byte("msg2"))
+		// Bob is not a member of the local MSP, policy check must fail
+		err = pc.CheckPolicyNoChannel(mgmt.Members, sProp)
+		require.Error(t, err)
+		require.Contains(t, err.Error(), "Failed deserializing proposal creator during channelless check policy with policy [Members]: [Invalid Identity]")
+	})
+
+	t.Run("CheckPolicyNoChannel", func(t *testing.T) {
```

Comment:
Nit: To differentiate, better to use different names, perhaps `CheckPolicyNoChannelSignedProposal` and `CheckPolicyNoChannelSignedData`. But this should not hold this PR from merging.